### ISSUE TITLE
Race buttons not working.

### ISSUE THREAD

**Issue by ArashdeepSaini** [oldnapalm/GTACoop#17]
The respawn, vote, leave and all other of these options are not working during race. Please help.
I configured the groups.xml file as you have mentioned. I am attaching the screenshots of both files (groups.xml and serverSettings.xml) Here is the screenshot of file. Please help. I want to know how do I get these options to work in game.
![image](https://github.com/user-attachments/assets/477dfd77-9d51-4a0c-9eea-575f7adc3ef4)
![image](https://github.com/user-attachments/assets/9d810894-d25f-4740-9ace-bcd3ace33ab0)

**Comment by oldnapalm** [comment]
Those are not buttons, they are commands you can run in the chat (press `T`).

Examples:
- `/vote canyon crossing` (can only be run when there's no active race (race finished or all players run `/leave`)
- `/join` (can be run by a player who connects to the server when a race have already started)
- `/respawn` (can only be run when participating in a race)

**Comment by ArashdeepSaini** [comment]
Okay thank you.
I have another issue, If I could get help here only. 
The ping of my friends is very high like 300 or 400. I am using ZeroTier network to get my friends join the server because only then it is possible for them to join otherwise they can't join. Tried Hamachi as well and I tried to do port forwarding on router settings by adding 4499 port in UDP and TCP as well but not able to get it working. Is there a way to make my server public at good ping? 
I would really appreciate your help as I am unable to find any solution.

**Comment by oldnapalm** [comment]
Ping depends on geographic distance between server and client, if your friends are from distant countries, their ping will be high, there's nothing you can do.

Public servers need to have the port exposed (UDP only, TCP is not needed). Although, some internet providers do not allow port forwarding (some allow if you contact them and ask to configure it), or it's possible that you have a more complex network setup, like if the router you are configuring is behind another firewall, then you need to forward the port in both.

Another possibility for a public server is using a VPS provider, like https://github.com/oldnapalm/GTACoop/wiki/Setup-a-server-on-Oracle-Cloud-Free-Tier

**Comment by ArashdeepSaini** [comment]
Can we add more maps of sky ramp races into this gamemodes folder from somewhere. I tried to add some but they didn't work. Any compatible maps other than these 151 maps?

**Comment by oldnapalm** [comment]
You can if you find or create other races in the [CommunityRaces](https://github.com/oldnapalm/CommunityRaces) format.

Instructions on how to create races can be found [here](https://www.gta5-mods.com/scripts/community-races-oldnapalm).

[This guy](https://github.com/oldnapalm/CommunityRaces/issues/5#issuecomment-1031575217) created a script to convert races from GTA Online, but it's been a while and I don't know if it still works.

**Comment by ArashdeepSaini** [comment]
Alright. I searched and check all the community races mods by you. What a great work sir. I have transferred the files into my scripts folder and it is working good. 
Just curious to know, how do I play these races with my buddies who play with my on GTACoop mod online? is it possible? I am sorry I am just a noob,,,,, my apologies if this is already answered somewhere. I tried to find it on google and YouTube but could find a way to do it. Thanks in advance.
